SwanContents/swancontents/filemanager/swanfilemanager.py

```diff
@@ -382,12 +382,13 @@ def download(self, url):
 
         if url.endswith('.git'):
 
-            rc = subprocess.call(['git', 'clone', url, tmp_dir_name])
+            # Set a timeout of 2 minutes for git clone to finish
+            rc = subprocess.run(['git', 'clone', '--', url, tmp_dir_name], timeout=120)
```

Comment:
Plus other commits that address @diocas comments.

Reply:
I would maybe add some of the info added in the commits to the code itself (so that we know why from reading the code). I'm specifically thinking on the `call` and `--`

Reply:
@PMax5 I believe you still need to address this last comment from @diocas .